plugins/chat/assets/javascripts/discourse/routes/chat-channel-thread.js

```diff
@@ -0,0 +1,18 @@
+import DiscourseRoute from "discourse/routes/discourse";
+import { inject as service } from "@ember/service";
+
+export default class ChatChannelThread extends DiscourseRoute {
+  @service router;
+  @service chatThreadsManager;
+  @service chatStateManager;
+  @service chat;
+
+  async model(params) {
+    return this.chatThreadsManager.find(params.threadId);
+  }
+
+  afterModel(model) {
+    this.chat.setActiveThread(model);
```

Comment:
I know we have `setActiveChannel` but we should change both to use setter/getter:

```javascript
this.chat.activeThread = model;
```

Reply:
I will make a secondary PR to change `activeChannel`, since it touches a lot more places